### PR TITLE
feat(RDS): add availability_zone filter

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
1. refactor to SDK
2. add availability_zone filter

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run=TestAccRdsFlavorDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run=TestAccRdsFlavorDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccRdsFlavorDataSource_basic
=== PAUSE TestAccRdsFlavorDataSource_basic
=== CONT  TestAccRdsFlavorDataSource_basic
--- PASS: TestAccRdsFlavorDataSource_basic (18.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       18.954s
```

```
make testacc TEST='./huaweicloud/services/acceptance/rds' TESTARGS='-run=TestAccRdsFlavorDataSource_all'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/rds -v -run=TestAccRdsFlavorDataSource_all -timeout 360m -parallel 4
=== RUN   TestAccRdsFlavorDataSource_all
=== PAUSE TestAccRdsFlavorDataSource_all
=== CONT  TestAccRdsFlavorDataSource_all
--- PASS: TestAccRdsFlavorDataSource_all (15.95s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/rds       16.036s
```

### CODE REVIEW

File: huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+	return nil
+}
+
+func flattenRdsFlavor(flavor flavors.Flavors, azList, versionList []string) map[string]interface{} {
+	vcpus, _ := strconv.Atoi(flavor.Vcpus)
+
+	return map[string]interface{}{
+		"id":                 flavor.ID,
+		"name":               flavor.Speccode,
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go:195]: is `name` equal to `id` ?

Comment (hmbig2) [reply huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go:195]: no, 
id is uuid
name like this: `rds.mysql.x1.4xlarge.2.rr`

File: huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	flavorsResp, err := flavors.ExtractDbFlavors(pages)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve RDS flavors: %s", err)
+	}
+
+	mode := d.Get("instance_mode").(string)
+	mem := d.Get("memory").(int)
+	az := d.Get("availability_zone").(string)
```
Comment (niuzhenguo) [inline huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go:127]: can we make `availability_zone` to be a list, maybe like `az1,az2,az3`

File: huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+		vcpus = strconv.Itoa(v.(int))
+	}
+
+	filter := map[string]interface{}{
+		"Vcpus":        vcpus,
+		"Ram":          mem,
+		"Instancemode": mode,
+	}
+	if mem, ok := d.GetOk("memory"); ok {
+		filter["Ram"] = mem.(int)
```
Comment (chengxiangdong) [inline huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go:139]: `filter["Ram"]` has setted at line 137, no need to set it up here

**Pull request merged**